lib/server.js

```diff
@@ -104,26 +117,31 @@ exports.create = function create(createOptions) {
       return done(new Error('asking to use react router for rendering, but no routes are provided'));
     }
 
-    var componentInstance;
+    var isRedirect = false;
 
     try {
       if (this.useRouter) {
-        match({ routes:createOptions.routes, location:thing},
```

Comment:
This is actually a potential asynchronous bug, which already exists in the (existing master branch)[https://github.com/paypal/react-engine/blob/master/lib/server.js#L114]. It is not safe to assume that a function that uses callbacks will be synchronous. If it is not, there is no guarantee that `componentInstance` will be populated before the later invocation of `ReactDOMServer.renderToString`. To fix this, I have refactored out the function `renderAndDecorate` and made all callbacks `return`.